lib/parser.js

```diff
@@ -202,28 +206,57 @@ async function customDocumentOperations(parsedJSON, asyncapiYAMLorJSON, initialF
   validateChannels(parsedJSON, asyncapiYAMLorJSON, initialFormat);
   validateOperationId(parsedJSON, asyncapiYAMLorJSON, initialFormat, OPERATIONS);
 
-  await customComponentsMsgOperations(parsedJSON, asyncapiYAMLorJSON, initialFormat, options);
-  await customChannelsOperations(parsedJSON, asyncapiYAMLorJSON, initialFormat, options);
+  await customComponentsMsgOperations(parsedJSON, asyncapiYAMLorJSON, initialFormat, options, schemaValidationErrors);
+  await customChannelsOperations(parsedJSON, asyncapiYAMLorJSON, initialFormat, options, schemaValidationErrors);
+
+  if(schemaValidationErrors.length) {
+    const errors = schemaValidationErrors.map(e => ({
+      ...e,
+      dataPath: e.dataPath.replace(/^\./, ''),
+    }));
+    schemaValidationErrors.length=0;
+    throw new ParserError({
+      type: validationError,
+      title: 'The document uses a schema format that parser does not support',
+      detail:'Parser is not able to parse messages as they are provided in not supported schema. Parser can be extended to parse other schemas, you can also use existing custom schema parsers for Avro, RAML Data Types and OpenAPI Schema. Install one of them, @asyncapi/avro-schema-parser, @asyncapi/openapi-schema-parser or @asyncapi/ramldt-schema-parser and use registerSchemaParser function to register it with the this parser',
+      parsedJSON,
+      validationErrors: errors,
+    });
+  }
```

Comment:
```suggestion
    if (schemaValidationErrors.size) {
    throw new ParserError({
      type: validationError,
      title: 'The document uses a schema format that parser does not support',
      detail: 'Parser is not able to parse messages as they are provided in not supported schema. Parser can be extended to parse other schemas, you can also use existing custom schema parsers for Avro, RAML Data Types and OpenAPI Schema. Install one of them, @asyncapi/avro-schema-parser, @asyncapi/openapi-schema-parser or @asyncapi/ramldt-schema-parser and use registerSchemaParser function to register it with the this parser',
      parsedJSON,
      validationErrors: groupValidationErrors(
        'channels',
        'uses unsupported schema format',
        schemaValidationErrors,
        asyncapiYAMLorJSON,
        initialFormat
      )
    });
  }
```

Reply:
This is only half the working code. This will now trigger good errors so you need to adjust tests accordingly

The only thing that doesn't work in this code is line 219 -> `'channels'`. It is hardcoded to channels even though sometimes as you know error might be not in message under channels but also in message under components, so sometimes instead of `'channels'` you must pass `'components'`.  

We need to figure out how to do this....so first please apply my suggestions, fix tests and what should left for us to think about is this channels vs components thingy. I think the only way to solve it will be to refactor `groupValidationErrors` somehow